subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -881,20 +850,282 @@ static int nrf_cloud_encode_service_info_ui(const struct nrf_cloud_svc_info_ui *
+{
+	char network_mode[12] = {0};
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE] = {0};
+	int ret;
+
+	__ASSERT_NO_MSG(network != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = add_modem_info_data(&network->current_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->sup_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->area_code, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->current_operator, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->ip_address, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->ue_mode, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_name_get(network->cellid_hex.type, data_name);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (cJSON_AddNumberToObjectCS(json_obj, data_name, network->cellid_dec) == NULL) {
+		return -EINVAL;
+	}
+
+	if (network->lte_mode.value == 1) {
+		strcat(network_mode, "LTE-M");
+	} else if (network->nbiot_mode.value == 1) {
+		strcat(network_mode, "NB-IoT");
+	}
+	if (network->gps_mode.value == 1) {
+		strcat(network_mode, " GPS");
+	}
+
+	if (cJSON_AddStringToObject(json_obj, "networkMode", network_mode) == NULL) {
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+static int encode_modem_info_sim(struct sim_param *sim, cJSON *json_obj)
+{
+	int ret;
+
+	__ASSERT_NO_MSG(sim != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = add_modem_info_data(&sim->uicc, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&sim->iccid, json_obj);
+	if (ret) {
+		LOG_DBG("sim_param object does not contain an ICCID");
+	}
+
+	ret = add_modem_info_data(&sim->imsi, json_obj);
+	if (ret) {
+		LOG_DBG("sim_param object does not contain an IMSI");
+	}
+
+	return 0;
+}
+
+static int encode_modem_info_device(struct device_param *device, cJSON *json_obj)
+{
+	int ret;
+
+	__ASSERT_NO_MSG(device != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = add_modem_info_data(&device->modem_fw, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&device->battery, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&device->imei, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	if (cJSON_AddStringToObject(json_obj, "board", device->board) == NULL) {
+		return -EINVAL;
+	}
+
+	if (cJSON_AddStringToObject(json_obj, "appVersion", device->app_version) == NULL) {
+		return -EINVAL;
+	}
+
+	if (cJSON_AddStringToObject(json_obj, "appName", device->app_name) == NULL) {
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+static int encode_modem_info_json_object(struct modem_param_info *modem,
+				  cJSON *root_obj)
+{
+	int ret;
+
+	__ASSERT_NO_MSG(root_obj != NULL);
+	__ASSERT_NO_MSG(modem != NULL);
+
+	cJSON *network_obj = NULL;
+	cJSON *sim_obj = NULL;
+	cJSON *device_obj = NULL;
+
+	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_NETWORK)) {
```

Comment:
From your comment here https://github.com/nrfconnect/sdk-nrf/pull/8304#discussion_r952368522

I think your intent was to remove these `IS_ENABLED(CONFIG_MODEM_INFO_ADD_XXX)` checks, right? (and rely on the `encode_modem_info_xxxx` functions to intelligently no-op based on info availability)  Or am I misunderstanding?

If your intent is to leave these checks in, it's not a problem, just want to make sure

Reply:
At least at the moment I don't see a good way to detect whether those parameters have been initialized or not.